zipline/pipeline/factors/factor.py

```diff
@@ -1205,17 +1205,22 @@ def compute(self, today, assets, out, close):
     '''
     dtype = float64_dtype
 
-    def __getattr__(self, attribute_name):
+    def __getattribute__(self, name):
+        outputs = object.__getattribute__(self, 'outputs')
+        if outputs is not NotSpecified and name in outputs:
+            return RecarrayField(factor=self, attribute=name)
+        else:
+            return super(CustomFactor, self).__getattribute__(name)
+
+    def __getattr__(self, name):
```

Comment:
This won't work exactly, because if outputs is specified we would be raising an AttributeError for all other attributes. The best way I can think of to do everything in `__getattribute__` is something like:

```python
if outputs is NotSpecified:
    return super(...).__getattribute__(...)
elif name in outputs:
    return RecarrayField(...)
else:
    try:
        return super(...).__getattribute__(...)
    except AttributeError:
        raise AttributeError("No such output named...")
```